control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -143,6 +144,16 @@ func (r *reconciler) reconcile(ctx context.Context) error {
 		return fmt.Errorf("failed to get hosted control plane %s/%s: %w", r.hcpNamespace, r.hcpName, err)
 	}
 
+	// Determine if the expected control plane operator image matches the active image in the hosted-cluster-config operator leader.
```

Comment:
Only need to access the index once:

expectedImage, ok := hostedControlPlane.Annotations[hyperv1.ControlPlaneOperatorImageAnnotation]
if !ok {
  return
}

if expectedImage != .ActiveImage {

Reply:
If we end up moving forward with current approach, we'll want this to be a unit tested function e.g `isCPOCompatibleWithHCP()`

Reply:
done